### PR TITLE
7714 component display name

### PR DESCRIPTION
**Pull request by carterbs**
**Resolves all non-app catalog-related tasks for [7714](https://chartiq.kanbanize.com/ctrl_board/18/cards/7714/details)**

**Needs this react-controls PR**: https://github.com/ChartIQ/finsemble-react-controls/pull/16
**Needs this finsemble PR**:  https://github.com/ChartIQ/finsemble/pull/1132

**Note**: The ticket said to use `components.foreign.['app launcher']`. Since the friendly name is being used in multiple places, I decided to put it on the 'component' property. Makes the most sense there, I think.

**Description of change**
-If the dev provides `config.component.friendlyName`, we use that in the title bar. If not, we use the default title provided in the config. If that doesn't exist, we use the window's name.
-If the dev provides `config.component.friendlyName`, we use that in the app launcher. If it isn't provided, we use the component config.
-If the dev provides `config.component.friendlyName`, we use that as the displayed pin. If the user changes the 

**Description of testing**
-Looked in App Launcher after updating friendly Name.
-Removed friendly name. Verified it didn't break.
-Added friendly name, verified it showed in title bar. Removed friendly name, verified that title property still worked.
-Pinned component. Restarted, changed friendly name. Was able to unpin component with new friendly name.

### CODE REVIEW

File: src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js [hunk 1]
```diff
@@ -45,7 +51,7 @@ var Actions = {
 			windowTitleBarStore.setValue({ field: "hackScrollbar", value: (windowTitleBarConfig.hackScrollbar !== false) });
 
 			// Set by calling WindowClient.setTitle() || from config "foreign.components.Window Manager.title"
-			var title = FSBL.Clients.WindowClient.title || windowTitleBarConfig.title;
+			var title = displayName || FSBL.Clients.WindowClient.title || windowTitleBarConfig.title;
```
[the remark this answers is not in the document]
Comment (kriswest) [reply src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js]: Sorry for delay responding. Title should have priority over display name in the title bar, displayname should have priority over title in the app catalog/launcher menu (IMHO). So yes, I agree with @thorsent 

However, one query, should we be matching new variable names to their FDC3 equivalents (for the sake of our sanity)? They have:
- name,
- appId,
- title,
- friendlyName (who picked that??)

friendlyName would be (I think) equivalent to displayName. 

See https://github.com/ChartIQ/fdc-appd/blob/11307_final_catalog_bugs/src/data.json for metadata examples (do look at several as they vary).

**Comment by amaroks** [comment]
@BradleyCarter Does this still need review?

**Comment by carterbs** [comment]
@amaroks not now

**Comment by carterbs** [comment]
@thorsent need you to weigh in on this. It's been languishing forever and somehow has no merge conflicts.


Should it be merged or closed?

**Comment by thorsent** [comment]
@carterbs I think if the items that Kris and I were discussing are resolved (we were in agreement with the ordering) then it's safe to merge. For the sake of getting it merged and closed I think we should punt on Kris' suggestion to match up with FDC3 param names.

**Comment by carterbs** [comment]
Pretty sure I changed the name thing a few months ago. Will confirm and
have someone retest.

File: src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js [hunk 2]
```diff
@@ -19,11 +19,11 @@ var Actions = {
 		let windowTitleBarConfig = options.customData.foreign.components["Window Manager"];
 		let FSBLHeader = windowTitleBarConfig.FSBLHeader;
 		let self = this;
-		let displayName = null;
+		let friendlyName = null;
 
 		//Display name is first up for display. If it doesn't exist, we look for a default title (or one that's already set on init). If no title, we display the window's name (e.g., '5434-Welcome Component-2321').
-		if (options.customData.component && options.customData.component.displayName) {
-			displayName = options.customData.component.displayName;
+		if (options.customData.component && options.customData.component.friendlyName) {
```
Comment (amaroks) [inline src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js]: Isn't there a chance that custom data is undefined?

**Comment by kriswest** [comment]
Chatted with @amaroks about this ticket. Since the last time it was worked on, the FDC3 appD spec has changed and dropped the friendlyName field. Theres also a bit of a mismatch between the overlapping fields, e.g.:
- **title** in Finsemble is the window title on the titlebar (which is closer to the HTML title tag than anything else)
    - in FDC3 its the display name in a launcher
- **name** in Finsemble is a unique ID for a window (appID + a GUID to make it unique)
    - in FDC3 its likely to be appID+version as it has to be unique except for versions of the same app

Hence, I suggested we go with whatever is least confusing (so displayName) and assume that we'll be mapping FDC3 fields to our own when working with an appD:
- fdc3.appID = finsemble.component name
- fdc3.title (or if not set fdc3.name or fdc3.appID) = finsemble.displayName
- fdc3.name = ignored
- fd3.title = finsemble. foreign.components."Window Manager".title

@thorsent does that make sense to you?

**Comment by amaroks** [comment]
I did some refactoring and made the following changes
- Fixed search results
- Search not only type/name but also displayName
- Use and give priority to displayName in appLauncher/Catalog

File: src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js [hunk 3]
```diff
@@ -45,7 +51,7 @@ var Actions = {
 			windowTitleBarStore.setValue({ field: "hackScrollbar", value: (windowTitleBarConfig.hackScrollbar !== false) });
 
 			// Set by calling WindowClient.setTitle() || from config "foreign.components.Window Manager.title"
-			var title = FSBL.Clients.WindowClient.title || windowTitleBarConfig.title;
+			var title = FSBL.Clients.WindowClient.title || windowTitleBarConfig.title || displayName;
```
Comment (zona-ciq) [inline src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js]:
```suggestion
			const title = FSBL.Clients.WindowClient.title || windowTitleBarConfig.title || displayName;
```

Comment (amaroks) [reply src-built-in/components/windowTitleBar/src/stores/windowTitleBarStore.js]: A very limited refactoring to some code. This is an old code that might be refactored one day but not at the moment. I'm keeping var as var and const as const and avoiding any rename.

File: src-built-in/components/appLauncher/src/components/componentList.jsx [hunk 4]
```diff
@@ -111,67 +111,70 @@ export default class appLauncherContainer extends React.Component {
...
 	}
+
+	render() {
+		return (
+			this.state.componentList
+				? this.renderComponentsList()
+				: <div></div>
+		);
+	}
 }
```
[the remark this answers is not in the document]
Comment (amaroks) [reply src-built-in/components/appLauncher/src/components/componentList.jsx]: I don't see it necessary and it just prints out state of component

**Comment by amaroks** [comment]
> There is one bug (probably not in the scope of this card): the search result menu won't make itself disappear when user click away, it will only disappear when user click on a search result item. We should probably fix this at some point. @BradleyCarter

This is an old one and was fixed long ago. it should go away after the merge

**Comment by meredith-ciq** [comment]
Couldn't fix the merge revert and remerge so made a new pr: https://github.com/ChartIQ/finsemble-seed/pull/399. This pr will likely be closed soon in favor of that one.

**Pull request closed without merging**